src/components/chart/tests/ChartTooltip.spec.tsx

```diff
@@ -0,0 +1,75 @@
+import {shallow} from 'enzyme';
+import * as React from 'react';
+import {DropPodPosition, IDropUIPosition} from '../../drop/DomPositionCalculator';
+
+import {DropPod} from '../../drop/DropPod';
+import {ChartTooltip} from '../ChartTooltip';
+import {ChartTooltipContent} from '../ChartTooltipContent';
+import {XYChartContextMock} from './XYChartContextMock';
+
+describe('<ChartTooltip />', () => {
+    beforeEach(() => {
+        spyOn(React, 'useContext').and.returnValue(XYChartContextMock);
+    });
+
+    it('should not throw', () => {
+        expect(() => {
+            shallow(<ChartTooltip />);
+            shallow(<ChartTooltip sort />);
+        }).not.toThrow();
+    });
+
+    it('should render two rectangles per data', () => {
+        const component = shallow(<ChartTooltip />);
+        expect(component.find('rect').length).toBe(XYChartContextMock.series[0].data.length * 2);
+    });
+
+    it('should render a DropPod', () => {
+        const component = shallow(<ChartTooltip />);
+        expect(component.find(DropPod).exists()).toBe(true);
+    });
+
+    it('should render a ChartTooltipContent inside the DropPod', () => {
+        const component = shallow(<ChartTooltip />);
+        const content = shallow(component.find(DropPod).prop('renderDrop')({}, React.createRef(), {position: DropPodPosition.left, orientation: ''}) as React.ReactElement);
+        expect(content.find(ChartTooltipContent).exists()).toBe(true);
+    });
+
+    it('should render a line when the user hover a serie point', () => {
+        const getAttributeSpy = jasmine.createSpy('getAttribute').and.returnValues(DropPodPosition.right, '10', '0');
+        const component = shallow(<ChartTooltip />);
+        const rect = component.find('rect').first();
+
+        rect.simulate('mouseMove', {currentTarget: {getAttribute: getAttributeSpy}});
+        // force update
+        component.setProps({sort: false});
+
+        // I can't seem to get this to work, the useState is never re-called

```

Comment:
Have you tried mounting the component instead of using shallow?

Reply:
Drop throws some errors because it can't detect the boundingBox of the element